### PR TITLE
Feature/menu

### PR DESCRIPTION
**Pull request by yurii-kdz**
added menu and content components

### CODE REVIEW

File: my-app/.idea/vcs.xml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project version="4">
```
Comment (ArtGarm) [inline my-app/.idea/vcs.xml:2]: move .idea files to gitignore

File: my-app/package.json [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  "name": "my-app",
+  "version": "0.1.0",
+  "private": true,
+  "dependencies": {
+    "@material-ui/core": "^4.11.0",
+    "@material-ui/lab": "^4.0.0-alpha.56",
+    "@material-ui/styles": "^4.10.0",
+    "@testing-library/jest-dom": "^4.2.4",
+    "@testing-library/react": "^9.3.2",
+    "@testing-library/user-event": "^7.1.2",
```
Comment (ArtGarm) [inline my-app/package.json]: we don't use it

File: my-app/package.json [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    "react": "^16.13.1",
+    "react-dom": "^16.13.1",
+    "react-scripts": "3.4.3"
+  },
+  "scripts": {
+    "start": "react-scripts start",
+    "build": "react-scripts build",
+    "test": "react-scripts test",
+    "eject": "react-scripts eject"
```
Comment (ArtGarm) [inline my-app/package.json]: this command useless for us now

File: my-app/src/App.js [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+    }
+})
+
+const App = () => {
+    const classes = useStyles();
+    return (
+        <div className={classes.root}>
+            <Header />
+            <Body />
+
```
Comment (ArtGarm) [inline my-app/src/App.js]: remove empty spacing

File: my-app/src/assets/img/arrow-up-circle.svg [hunk 5]
```diff
@@ -0,0 +1,5 @@
+<svg width="24" height="24" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg">
```
Comment (ArtGarm) [inline my-app/src/assets/img/arrow-up-circle.svg]: refactor it to SvgIcon

File: my-app/src/assets/img/home.svg [hunk 6]
```diff
@@ -0,0 +1,4 @@
+<svg width="24" height="24" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg">
```
Comment (ArtGarm) [inline my-app/src/assets/img/home.svg:1]: refactor it to SvgIcon

File: my-app/src/components/Body/Content/BlogView/BlogView.jsx [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+    return {name, posted, status, stats, views, dots};
+};
+
+const rows = [
+    createData('Design: A Survival Guide for Beginners', 'Posted 3 days ago', "Scheduled", 120, "views", ""),
+    createData('Design: A Survival Guide for Beginners', 'Posted 3 days ago', 'Draft', 120, "views", ""),
+    createData('Design: A Survival Guide for Beginners', 'Posted 3 days ago', 'Scheduled', 120, "views", ""),
+    createData('Design: A Survival Guide for Beginners', 'Posted 3 days ago', 'Published', 0, "views", ""),
+    createData('Design: A Survival Guide for Beginners', 'Posted 3 days ago', 'Published', 120, "views", ""),
+];
```
Comment (ArtGarm) [inline my-app/src/components/Body/Content/BlogView/BlogView.jsx]: useless function

if you want to create mock data - replace it to separate file and describe there

File: my-app/src/components/Body/Content/BlogView/BlogView.jsx [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+                            <TableCell className={classes.tableCell} align="left">Title</TableCell>
+                            <TableCell className={classes.tableCell} align="left">Status</TableCell>
+                            <TableCell className={classes.tableCell} align="left">Stats</TableCell>
+                            <TableCell className={classes.tableCell} align="right">
+                                <Button startIcon={<PenToolWhite/>} className={classes.button}>Add new</Button>
+                            </TableCell>
+                        </TableRow>
+                    </TableHead>
+                    <TableBody>
+                        {rows.map((row) => (
```
Comment (ArtGarm) [inline my-app/src/components/Body/Content/BlogView/BlogView.jsx]: use destruction for loop param

File: my-app/src/components/Body/Content/BlogView/BlogView.jsx [hunk 9]
```diff
@@ -0,0 +1,90 @@
...
+                                <TableCell component="th" scope="row" className={classes.cell}>
+                                    <Typography variant="h3" className={classes.name}>
+                                        {row.name}
+                                    </Typography>
+                                    <Typography variant="h5" className={classes.subtext}>
+                                        {row.posted}
+                                    </Typography>
+                                </TableCell>
+                                <TableCell align="center" className={classes.cell}>
+                                <span className={classes.status + " " + row.status}>
```
Comment (ArtGarm) [inline my-app/src/components/Body/Content/BlogView/BlogView.jsx]: what do you want here??
newer concat classes

File: my-app/src/components/Body/Content/BlogView/BlogView.jsx [hunk 10]
```diff
@@ -0,0 +1,90 @@
...
+                                        <span className={classes.stats}>{row.stats}</span>
+                                        <span className={classes.views}>{row.views}</span>
+                                        {row.stats > 0
+                                            ? <span> <img src={arrow} alt="arrow"/></span>
+                                            : <span> </span>
+                                        }
+                                    </div>
+                                </TableCell>
+                                <TableCell align="right" className={classes.cell}>
+                                    <div className={classes.dots}>
```
Comment (ArtGarm) [inline my-app/src/components/Body/Content/BlogView/BlogView.jsx]: use already exist component for this dots or create own

File: my-app/src/components/Body/Content/BlogView/BlogView.jsx [hunk 11]
```diff
@@ -0,0 +1,90 @@
...
+                </Table>
+            </TableContainer>
+            <Pagination count={3} hidePrevButton hideNextButton color="primary" classes={{ul: classes.ul}}
+                        className={classes.pagination}/>
+        </div>
+
+    );
+}
+
+export default BlogView
```
Comment (ArtGarm) [inline my-app/src/components/Body/Content/BlogView/BlogView.jsx:38]: try to make smaller components

File: my-app/src/components/Body/Content/BlogView/style.js [hunk 12]
```diff
@@ -0,0 +1,123 @@
...
+    ul: {
+        "& li" : {
+            "& button" : {
+                background: "#e2e8f0",
+                color: "#2c5282",
+                fontWeight: 700
+            }
+        }
+    }
+});
```
Comment (ArtGarm) [inline my-app/src/components/Body/Content/BlogView/style.js]: can we import some from theme?

File: my-app/src/components/Body/Content/Content.jsx [hunk 13]
```diff
@@ -0,0 +1,15 @@
+import React from "react";
+import BlogView from "./BlogView/BlogView";
+import useStyles from "./style";
+
+const Content = () => {
+    const classes = useStyles();
+
+    return(
+        <div className={classes.content}>
```
Comment (ArtGarm) [inline my-app/src/components/Body/Content/Content.jsx]: maybe try to use Grid?

File: my-app/src/components/Body/Menu/Menu.jsx [hunk 14]
```diff
@@ -0,0 +1,71 @@
...
+    {text: "Users", icon: <Users/>},
+    {text: "Subscriptions", icon: <Zap/>},
+    {text: "Archived pages", icon: <Trash/>}
+];
+
+const menuItemsSecondary = [
+    {text: "Themes", icon: <Book/>},
+    {text: "Plugins", icon: <Box/>},
+    {text: "Update plans", icon: <Coffee/>},
+];
```
Comment (ArtGarm) [inline my-app/src/components/Body/Menu/Menu.jsx]: replace mock data to separate file

File: my-app/src/components/Body/Menu/Menu.jsx [hunk 15]
```diff
@@ -0,0 +1,71 @@
...
+
+const Menu = () => {
+    const classes = useStyles();
+
+    let menuItem = menuItems.map(i =>
+        <ListItem className={classes.listItem}>
+            {i.icon}
+            <ListItemText className={classes.item}
+                          primary={i.text}
+            />
```
Comment (ArtGarm) [inline my-app/src/components/Body/Menu/Menu.jsx]: use eslint for this

File: my-app/src/components/Body/Menu/Menu.jsx [hunk 16]
```diff
@@ -0,0 +1,71 @@
...
+    {text: "Themes", icon: <Book/>},
+    {text: "Plugins", icon: <Box/>},
+    {text: "Update plans", icon: <Coffee/>},
+];
+
+const Menu = () => {
+    const classes = useStyles();
+
+    let menuItem = menuItems.map(i =>
+        <ListItem className={classes.listItem}>
```
Comment (ArtGarm) [inline my-app/src/components/Body/Menu/Menu.jsx]: use css api

File: my-app/src/components/Body/Menu/style.js [hunk 17]
```diff
@@ -0,0 +1,38 @@
+import {makeStyles} from "@material-ui/styles";
```
Comment (ArtGarm) [inline my-app/src/components/Body/Menu/style.js:1]: same issue
import some from theme

File: my-app/src/components/Body/Body.jsx [hunk 18]
```diff
@@ -0,0 +1,17 @@
+import React from "react";
+import Menu from "./Menu/Menu";
+import useStyles from "./style";
+import Content from "./Content/Content";
+
+const Body = () => {
+    const classes = useStyles();
+
+    return(
+        <div className={classes.body}>
```
Comment (ArtGarm) [inline my-app/src/components/Body/Body.jsx]: use Grid

File: my-app/src/components/Header/style.js [hunk 19]
```diff
@@ -0,0 +1,32 @@
+import {makeStyles} from "@material-ui/styles";
+
+const useStyles = makeStyles({
```
Comment (ArtGarm) [inline my-app/src/components/Header/style.js]: same issue

File: my-app/src/icons/Zap.js [hunk 20]
```diff
@@ -0,0 +1,13 @@
+import React from 'react';
```
Comment (ArtGarm) [inline my-app/src/icons/Zap.js:1]: Do icons need some props?

**Review by ArtGarm — CHANGES_REQUESTED** [review]

File: my-app/src/components/Body/Content/BlogView/BlogView.jsx [hunk 21]
```diff
@@ -0,0 +1,38 @@
...
+                            <TableCell className={classes.tableCell} align="left">Stats</TableCell>
+                            <TableCell className={classes.tableCell} align="right">
+                                <Button startIcon={<PenToolWhite/>} size="large" >Add new</Button>
+                            </TableCell>
+                        </TableRow>
+                    </TableHead>
+                    <BlogViewTableBody />
+                </Table>
+            </TableContainer>
+            <Pagination count={3} hidePrevButton hideNextButton color="primary" classes={{ul: classes.ul}} className={classes.pagination}/>
```
Comment (ArtGarm) [inline my-app/src/components/Body/Content/BlogView/BlogView.jsx:33]: is it need classes.ul and classes.pagination

File: my-app/src/components/Body/Content/BlogView/BlogViewTableBody/BlogViewTableBody.jsx [hunk 22]
```diff
@@ -0,0 +1,49 @@
...
+                        </Grid>
+                    </TableCell>
+                    <TableCell align="center" className={classes.cell}>
+                        <Grid container alignItems="center"className={classes.container}>
+                            <span className={classes.stats}>{row.stats}</span>
+                            <span className={classes.views}>{row.views}</span>
+                            {row.stats > 0
+                                ? <ArrowUp/>
+                                : <span> </span>
+                            }
```
Comment (ArtGarm) [inline my-app/src/components/Body/Content/BlogView/BlogViewTableBody/BlogViewTableBody.jsx]: empty span?

File: my-app/src/components/Body/Content/BlogView/BlogViewTableBody/style.js [hunk 23]
```diff
@@ -0,0 +1,46 @@
...
+const useStyles = makeStyles( theme =>({
+    root: {
+        borderRadius: "50px",
+        width: "120px",
+        height: "32px",
+        textTransform:'capitalize',
+    },
+    scheduled: {
+        color: "#F6AD55",
+        border: "2px solid #F6AD55",
```
Comment (ArtGarm) [inline my-app/src/components/Body/Content/BlogView/BlogViewTableBody/style.js]: get colors from theme

File: my-app/src/components/Header/style.js [hunk 24]
```diff
@@ -0,0 +1,13 @@
...
+    header: {
+        backgroundColor: '#2D3748',
+        padding: '24px 26px 19px 75px',
```
Comment (ArtGarm) [inline my-app/src/components/Header/style.js]: use theme spacing

**Review by ArtGarm — CHANGES_REQUESTED** [review]

File: my-app/package.json (hunk 2 above)
Comment (yurii-kdz) [reply my-app/package.json]: done

File: my-app/package.json (hunk 3 above)
Comment (yurii-kdz) [reply my-app/package.json]: done

File: my-app/src/App.js (hunk 4 above)
Comment (yurii-kdz) [reply my-app/src/App.js]: done

File: my-app/src/assets/img/arrow-up-circle.svg (hunk 5 above)
Comment (yurii-kdz) [reply my-app/src/assets/img/arrow-up-circle.svg]: done

File: my-app/src/assets/img/home.svg (hunk 6 above)
Comment (yurii-kdz) [reply my-app/src/assets/img/home.svg:1]: done

File: my-app/src/components/Body/Content/BlogView/BlogView.jsx (hunk 7 above)
Comment (yurii-kdz) [reply my-app/src/components/Body/Content/BlogView/BlogView.jsx]: done

File: my-app/src/components/Body/Content/BlogView/BlogView.jsx (hunk 8 above)
Comment (yurii-kdz) [reply my-app/src/components/Body/Content/BlogView/BlogView.jsx]: done

File: my-app/src/components/Body/Content/BlogView/BlogView.jsx (hunk 9 above)
Comment (yurii-kdz) [reply my-app/src/components/Body/Content/BlogView/BlogView.jsx]: done

File: my-app/src/components/Body/Content/BlogView/BlogView.jsx (hunk 10 above)
Comment (yurii-kdz) [reply my-app/src/components/Body/Content/BlogView/BlogView.jsx]: done (create own)

File: my-app/src/components/Body/Content/BlogView/BlogView.jsx (hunk 11 above)
Comment (yurii-kdz) [reply my-app/src/components/Body/Content/BlogView/BlogView.jsx:38]: done

File: my-app/src/components/Body/Content/Content.jsx (hunk 13 above)
Comment (yurii-kdz) [reply my-app/src/components/Body/Content/Content.jsx]: done

File: my-app/src/components/Body/Body.jsx (hunk 18 above)
Comment (yurii-kdz) [reply my-app/src/components/Body/Body.jsx]: done

File: my-app/src/icons/Zap.js (hunk 20 above)
Comment (yurii-kdz) [reply my-app/src/icons/Zap.js:1]: done

File: my-app/src/components/Header/style.js (hunk 24 above)
Comment (yurii-kdz) [reply my-app/src/components/Header/style.js]: done

File: my-app/src/components/Body/Content/BlogView/BlogViewTableBody/style.js (hunk 23 above)
Comment (yurii-kdz) [reply my-app/src/components/Body/Content/BlogView/BlogViewTableBody/style.js]: done

File: my-app/src/components/Body/Content/BlogView/BlogView.jsx (hunk 21 above)
Comment (yurii-kdz) [reply my-app/src/components/Body/Content/BlogView/BlogView.jsx:33]: done

File: my-app/src/components/Body/Menu/Menu.jsx (hunk 14 above)
Comment (yurii-kdz) [reply my-app/src/components/Body/Menu/Menu.jsx]: done

File: my-app/src/components/Body/Menu/Menu.jsx (hunk 16 above)
Comment (yurii-kdz) [reply my-app/src/components/Body/Menu/Menu.jsx]: done

File: my-app/src/components/Body/Content/BlogView/BlogViewTableBody/BlogViewTableBody.jsx (hunk 22 above)
Comment (yurii-kdz) [reply my-app/src/components/Body/Content/BlogView/BlogViewTableBody/BlogViewTableBody.jsx]: done

File: my-app/src/components/Body/Content/BlogView/BlogView.jsx [hunk 25]
```diff
@@ -0,0 +1,38 @@
...
+import PenToolWhite from "../../../../icons/PenToolWhite";
+import BlogViewTableBody from "./BlogViewTableBody/BlogViewTableBody";
+
+const BlogView = (props) => {
+    const classes = useStyles(props);
+
+    return (
+        <Grid container justify="center">
+            <TableContainer classes={{root: classes.root}} component={Grid}>
+                <Table className={classes.table} aria-label="simple table">
```
Comment (ArtGarm) [inline my-app/src/components/Body/Content/BlogView/BlogView.jsx]: is it need aria-label??

File: my-app/src/components/Body/Content/BlogView/BlogViewTableBody/style.js [hunk 26]
```diff
@@ -0,0 +1,45 @@
...
+        color: theme.palette.error.main,
+        marginRight: theme.spacing(26),
+    },
+    published: {
+        color: theme.palette.success.main,
+        marginRight: theme.spacing(26),
+    },
+    views: {
+        fontSize: "14px",
+        color: "#A0AEC0",
```
Comment (ArtGarm) [inline my-app/src/components/Body/Content/BlogView/BlogViewTableBody/style.js]: import from theme

File: my-app/src/components/Body/Menu/Menu.jsx [hunk 27]
```diff
@@ -0,0 +1,41 @@
...
+        </ListItem>
+    );
+
+    let menuItemSecondary = data.menuItemsSecondary.map(i =>
+        <ListItem className={classes.listItem}>
+            {i.icon}
+            <ListItemText classes={{primary: classes.primary}} primary={i.text}/>
+        </ListItem>
+    )
+
```
Comment (ArtGarm) [inline my-app/src/components/Body/Menu/Menu.jsx]: these functions the same

File: my-app/src/services/enum.js [hunk 28]
```diff
@@ -0,0 +1,10 @@
+const views = {
+    TOP : "120",
+    BOTTOM: "0",
+};
+
+let view = views.TOP;
+
+if(!view) {
+    throw new Error ("View is not defined!");
+}
```
Comment (ArtGarm) [inline my-app/src/services/enum.js]: remove this and create simple object

File: my-app/src/services/proxy.js [hunk 29]
```diff
@@ -0,0 +1,16 @@
...
+}
+
+const wrap = obj => {
+    return new Proxy(obj, {
+        get(target, propKey) {
+            console.log(`Reading property "${propKey}"`)
+            return target[propKey]
+        }
+    })
+}
```
Comment (ArtGarm) [inline my-app/src/services/proxy.js]: or finish this

File: my-app/src/services/proxy.js [hunk 30]
```diff
@@ -0,0 +1,16 @@
...
+const object = { message: 'never mind i found someone like you!' }
+const wrapped = wrap(object)
+console.log(wrapped.message)
```
Comment (ArtGarm) [inline my-app/src/services/proxy.js]: remove console log

for send error to console use console error

**Review by ArtGarm — CHANGES_REQUESTED** [review]

File: my-app/src/themes/theme.js [hunk 31]
```diff
@@ -0,0 +1,107 @@
...
+                fontSize: "16px",
+                lineHeight: "19px",
+                color: "#2C5282",
+                padding: "16px 0 26px 0"
+            }
+        }
+    }
+});
+
+export default theme;
```
Comment (ArtGarm) [inline my-app/src/themes/theme.js:107]: split theme 
3 parts
typography, palette, overrides

**Review by ArtGarm — APPROVED** [review]

**Review by ArtReactDev — APPROVED** [review]

**Pull request merged**